### PR TITLE
ci: Move to stable virtiofs

### PR DESCRIPTION
**Pull request by jcvenegas**
Create a virtiofs job to explicitly export
variables for virtiofs testing, like qemu, kernel,
and test to run.

Add a variable shared_fs to define what shared fs will be used
virtiofs could run even witout qemu_experimental.

Fixes: github.com/kata-containers/runtime#3083

Depends-on: github.com/kata-containers/runtime#3122
Depends-on: github.com/kata-containers/packaging#1190

Signed-off-by: Carlos Venegas <jos.c.venegas.munoz@intel.com>

### CODE REVIEW

**Review by jodh-intel — APPROVED** [review]

**Review by devimc — APPROVED** [review]
thanks @jcvenegas

File: .ci/jenkins_job_build.sh [hunk 1]
```diff
@@ -46,6 +46,11 @@ init_ci_flags() {
 	# Use experimental kernel
 	# Values: true|false
 	export experimental_kernel="false"
+	# Use experimental qemu
+	export experimental_qemu="true"
+	# shared fs to use
+	# Values: 9pfs|virtiofs
+	export shared_fs="9pfs"
```
Comment (fidencio) [inline .ci/jenkins_job_build.sh:53]: @jcvenegas, my OCD is killing me.

Would be possible to have those two newly added in capital letters, just to match with the majority of the other ones we already have?
Then, in a separate commit, would be possivel to also switch experimental_kernel to all caps? :-)

Apart from that, lgtm!

File: .ci/install_runtime.sh [hunk 2]
```diff
@@ -49,7 +49,7 @@ PKGDEFAULTSDIR="${SHAREDIR}/defaults/kata-containers"
 NEW_RUNTIME_CONFIG="${PKGDEFAULTSDIR}/configuration.toml"
 # Note: This will also install the config file.
 build_and_install "${runtime_repo}" "" "true" "${tag}"
-experimental_qemu="${experimental_qemu:-false}"
+shared_fs="${shared_fs:-9pfs}"
```
Comment (wainersm) [inline .ci/install_runtime.sh:52]: I think this ability to switch between 9p and virtiofs would be useful to 2.x too. For example, I just set a commit (https://github.com/kata-containers/tests/pull/3175/commits/0f1b6d915774ba5b0dec128855e7bef796bff4c0) that changes the shared_fs when running on openshift-ci. What you proposing here is much better.

File: .ci/jenkins_job_build.sh (hunk 1 above)
Comment (jcvenegas) [reply .ci/jenkins_job_build.sh:53]: Sure :) make sense

File: .ci/install_runtime.sh (hunk 2 above)
Comment (jcvenegas) [reply .ci/install_runtime.sh:52]: @wainersm I see , I think we could have something like this on 2.0 that allows to propagate the shared backend to the kata-runtime config. So we can explicitly what the openshift  should be testing.

**Comment by fidencio** [comment]
I'm really sorry for the disruption but in order to do the branch name changes announced [here](http://lists.katacontainers.io/pipermail/kata-dev/2021-January/001695.html) I'm adding the "do-not-merge" and the "needs-branch-adjustment" labels to the PR.

I'll add a new comment once the branch changes are done and once it happens I'd like to ask you to resubmit the PR against the new `main` branch, possibly linking to this PR in case you find the conversations / reviews already made important to proceed.

**Comment by fidencio** [comment]
Kata Containers repos are back to business.  Thanks a lot for your patience and I'm very happy to announce that there's no changes needed from your side as renaming the branches automagically reflected here.

If you notice something weird, some missing CI, please, do not hesitate on contacting us here or on #kata-dev on bit.ly/katacontainersslack.

Also, this is a good opportunity to have your PR rebased atop of `main`, to ensure your PR follow all the guidelines (https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format) enforced by this community, or even to re-evaluate whether the PR is currently needed.

Last but not least,  as the default branch has been renamed to `main`, if you have a local clone you'd need to update it and you do so by running:
```
git branch -m 2.0-dev main
git fetch origin
git branch -u origin/main main
```

Happy hacking!

**Comment by fidencio** [comment]
Oops, this was opened against `master`, please, ignore the previous messages. :-)

**Comment by jcvenegas** [comment]
For virtiofs jobs the next test are failing

`tests/conformance/posixfs`
````
17:07:43 Test Summary Report
17:07:43 -------------------
17:07:43 ./tests/chmod/12.t          (Wstat: 0 Tests: 14 Failed: 6)
17:07:43   Failed tests:  3-4, 7-8, 11-12
17:07:43 ./tests/link/00.t           (Wstat: 0 Tests: 202 Failed: 10)
17:07:43   Failed tests:  135-136, 142-143, 149-150, 156-157, 163-164
17:07:43 ./tests/mkdir/00.t          (Wstat: 0 Tests: 36 Failed: 2)
17:07:43   Failed tests:  33-34
17:07:43 ./tests/mkfifo/00.t         (Wstat: 0 Tests: 36 Failed: 2)
17:07:43   Failed tests:  33-34
17:07:43 ./tests/mknod/00.t          (Wstat: 0 Tests: 36 Failed: 2)
17:07:43   Failed tests:  33-34
17:07:43 ./tests/mknod/11.t          (Wstat: 0 Tests: 28 Failed: 4)
17:07:43   Failed tests:  12-13, 25-26
17:07:43 ./tests/open/00.t           (Wstat: 0 Tests: 47 Failed: 2)
17:07:43   Failed tests:  33-34
17:07:43 ./tests/utimensat/08.t      (Wstat: 0 Tests: 9 Failed: 2)
17:07:43   Failed tests:  5-6
```

File: .ci/install_qemu_experimental.sh [hunk 3]
```diff
@@ -21,8 +21,10 @@ CURRENT_QEMU_TAG=$(get_version "assets.hypervisor.qemu-experimental.tag")
 QEMU_TAR="kata-static-qemu-virtiofsd.tar.gz"
 arch=$("${cidir}"/kata-arch.sh -d)
 QEMU_PATH="${DESTDIR:-}/opt/kata/bin/qemu-virtiofs-system-x86_64"
-VIRTIOFS_PATH="${DESTDIR:-}/opt/kata/bin/virtiofsd"
 bindir="${DESTDIR:-}/usr/bin"
+DEST_QEMU_PATH="${bindir}/qemu-system-x86_64"
```
Comment (jodh-intel) [inline .ci/install_qemu_experimental.sh:25]: Can you create a `uname -m` variable to avoid hard-coding the arch here?

**Comment by jcvenegas** [comment]
All passing, but the PR in runtime has some errors on metrics, even when it is passing here, adding do-not-merge label

**Pull request merged**